### PR TITLE
Move towards olympia.test domain for local development

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This changes everything I found so far.

Fixes mozilla/addons#493

Open todos:

 * We need to update something on our fxa endpoints so that olympia.test is allowed there, I couldn't get log-in to work locally
 * addons-frontend needs to be aware of this change cc @kumar303 @tofumatt 

Anything else I'm missing?

cc @diox

### CODE REVIEW

**Comment by kumar303** [comment]
For local frontend development we use a proxy (with [`yarn amo:dev`](https://github.com/mozilla/addons-frontend/#running-amo-for-local-development)) to simulate a same-origin setup like prod; the proxy runs our local JS code against an API hosted on dev or stage. This means the log-in process will bounce to FxA and then redirect to the absolute URL which has been hard-coded for each respective [FxA client](https://oauth-stable.dev.lcip.org/console/login).

Here are the clients we depend on:

* This is for when we proxy to dev
  * https://github.com/mozilla/addons-server/blob/master/src/olympia/conf/dev/settings.py#L238
* This is for when we proxy to stage
  * https://github.com/mozilla/addons-server/blob/master/src/olympia/conf/stage/settings.py#L234
* This is for when we run our own local instance of addons-server via docker
  * https://github.com/mozilla/addons-server/blob/master/settings.py#L116

You have to ssh into dev or stage to see the value of the client ID by looking up the environment variable. This is annoying because it's not a secret value.

For each of these client configurations, **the `redirect_url` parameter is ignored and does nothing**. We should delete that key. The only way to change the `redirect_url` is to log in to the [FxA console](https://oauth-stable.dev.lcip.org/console/login) and update it (providing you have access!).

When we run our frontend proxy, we set `fxaConfig` to [`local`](https://github.com/mozilla/addons-frontend/blob/master/config/development-amo.js#L11) which is how it tells the proxied API which client ID and redirect URL to use.

Please don't break any of this 😃 I can help test it, just let me know. If at all possible, perhaps you can leave the `local` configuration in tact and add a new key for the new olympia domain until we know it's fully working. I'm thinking you might have to break the docker one for us. That's probably fine. We don't use it very much.

**Comment by kumar303** [comment]
Oh, wait, I think the only config that applies to the frontend's usage of `olympia.dev` is https://github.com/mozilla/addons-server/blob/master/settings.py#L116

You can probably ignore the others, my bad.

**Comment by EnTeQuAk** [comment]
> For each of these client configurations, the redirect_url parameter is ignored and does nothing. We should delete that key. 

I will do that right away then, this is just confusing otherwise. A comment should be more reasonable.

> The only way to change the redirect_url is to log in to the FxA console and update it (providing you have access!).

I don't unfortunately. I do remember that you were able to configure one or multiple allowed redirect_urls, is that still the case? Maybe @muffinresearch has access?

**Comment by kumar303** [comment]
I forget what the client IDs are but if you send some to me, I could try. I don't remember ever having access. I think we asked Mark Striemer last time, heh. We could ask him again if needed.

**Comment by mstriemer** [comment]
You can create a new config in the FxA console and set its redirect URL to http://olympia.test. That's probably easiest and should allow both URLs to be used if you're running an old version of addons-server without these changes.

If you'd rather update the redirect URL of the old config I can make that change, just let me know when.

Our `redirect_url` is indeed not used. I just copy/pasted those settings from the blob you get when you create the token and thought it might be convenient to be able to see what URL it redirects to, but since it can be edited there's no guarantee that it's correct and is just confusing as mentioned above.

**Comment by EnTeQuAk** [comment]
Closing for a moment, I'll get back to that asap once my other PRs are done.

**Comment by jrbenny35** [comment]
Can this be reopened?

**Comment by EnTeQuAk** [comment]
The PR will be once me or someone else has time to tackle this all
together with addons frontend. I might be able to make time some
time tomorrow

On Mon, Feb 5, 2018, at 7:26 PM, Benjamin Forehand Jr wrote:
> Can this be reopened?


> — You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].> 

Links:

  1. https://github.com/mozilla/addons-server/pull/7402#issuecomment-363175174

**Comment by eviljeff** [comment]
I created a new oauth client on fxa and updated the settings.py.
We should probably tidy all that up in follow-up (internal is unused; redirect_url is ignored; local is... no idea)?

**Comment by vladikoff** [comment]
@eviljeff let me know if you still need updates to the old one.

File: Makefile-docker [hunk 1]
```diff
@@ -141,9 +141,9 @@ ui-tests:
 	python manage.py update_permissions_from_mc
 
 	# Create a proper superuser that can be used to access the API
-	python manage.py waffle_switch super-create-accounts on
-	python manage.py waffle_switch activate-autograph-signing on
-	python manage.py createsuperuser --email=uitest@mozilla.com --username=uitest --noinput --add-to-supercreate-group --save-api-credentials=tests/ui/variables.json --hostname=olympia.dev
+	python manage.py waffle_switch super-create-accounts on --create
```
Comment (eviljeff) [inline Makefile-docker]: Did you move these waffle changes to another pull?  (Or did we just talk about it)

Comment (EnTeQuAk) [reply Makefile-docker]: I think we only talked about it, I'll back them out and commit them in a separate PR

**Comment by kumar303** [comment]
I tried this out for addons-frontend development and it's working for me after this small config change https://github.com/mozilla/addons-frontend/pull/4349

**Pull request merged**